components/ai_chat/content/browser/ai_chat_tab_helper.cc

```diff
@@ -345,6 +389,58 @@ void AIChatTabHelper::CheckPDFA11yTree() {
   OnPDFA11yInfoLoaded();
 }
 
+void AIChatTabHelper::FetchSearchQuerySummary(
+    FetchSearchQuerySummaryCallback callback) {
+  if (!IsBraveSearchSERP(GetPageURL())) {
+    std::move(callback).Run(std::nullopt);
+    return;
+  }
+
+  auto internal_callback =
+      base::BindOnce(&AIChatTabHelper::OnSearchSummarizerKeyFetched,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback));
+  GetSearchSummarizerKey(web_contents(), std::move(internal_callback));
+}
+
+void AIChatTabHelper::OnSearchSummarizerKeyFetched(
+    FetchSearchQuerySummaryCallback callback,
+    const std::optional<std::string>& key) {
+  if (!key) {
+    std::move(callback).Run(std::nullopt);
+    return;
+  }
+
+  if (!api_request_helper_) {
+    api_request_helper_ =
+        std::make_unique<api_request_helper::APIRequestHelper>(
+            GetSearchQuerySummaryNetworkTrafficAnnotationTag(),
+            web_contents()
+                ->GetBrowserContext()
+                ->GetDefaultStoragePartition()
+                ->GetURLLoaderFactoryForBrowserProcess());
+  }
+
+  GURL base_url("https://search.brave.com/api/chatllm/raw_data");
```

Comment:
I'm thinking we should use `brave_domains::GetServicesDomain("search")` for this instead of hard-coding the hostname.

Reply:
Addressed in https://github.com/brave/brave-core/pull/25257/commits/b17b5128d6f3b7fd82477e6070095ac2190b8549